### DISCUSSION TITLE
Azure Resource Manager using Workload Identity federation

### DISCUSSION THREAD

**Discussion by sannicolao** [Azure-Player/azure.datafactory.tools#408] [Show and tell]
**Describe the bug**
During Publish ADF i receive the error: Unsupported authentication scheme 'WorkloadIdentityFederation' for Azure endpoint.

**To Reproduce**
![image](https://github.com/user-attachments/assets/bc96de4c-9cb9-42cd-bf84-10c28ebb821c)

![image](https://github.com/user-attachments/assets/2a7e80f1-c5f6-4066-9814-33c29754d093)

With Service Principal, it works very well. With "Azure Resource Manager using Workload Identity federation" i receive the error. 

Thanks. 

Installed Version: 
![image](https://github.com/user-attachments/assets/c59f1d58-abaf-4d63-a3ad-fe5786398d5a)

**Comment by NowinskiK** [comment]
Change version from 1.x to **2.x** to get support for WIF:
![image](https://github.com/user-attachments/assets/d9f2da15-3930-4c16-9767-43f68c8a0947)

**Reply by dvartanian** [reply]
@NowinskiK 
We've just made the switch to federated service connections and so I too have updated my yaml to use the v2 version. Unfortunately, I'm now getting an error:
```
##[error]You cannot call a method on a null-valued expression.
```

This appears to be happen when it attempts to get the federated service connection's token:
```
##[debug]Get-VstsFederatedToken : You cannot call a method on a null-valued expression.
##[debug]At C:\a\_tasks\PublishADFTask_1af843b5-35a0-411f-9a18-9eb7a59fb8b8\2.0.1514\ps_modules\VstsAzureHelpers_\InitializeAzModuleFunctions.ps1:327 char:31
##[debug]+ ... ertionJwt = Get-VstsFederatedToken -serviceConnectionId $connectedSer ...
##[debug]+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##[debug]    + CategoryInfo          : InvalidOperation: (:) [Get-VstsFederatedToken], RuntimeException
##[debug]    + FullyQualifiedErrorId : InvokeMethodOnNull,Get-VstsFederatedToken
```

I have supplied that service connection name in the task inputs as shown below:
![image](https://github.com/user-attachments/assets/5cccb926-accb-40e8-a685-0810a4ac5577)
Is there something else I'm missing that might be causing this?

**Comment by sannicolao** [comment]
@NowinskiK Oh dam.... it works perfectly. Thanks.

**Reply by fbglv** [reply]
@NowinskiK I was wondering: are you using the (old) Release Pipelines or the "normal" ones?

**Reply by NowinskiK** [reply]
Do you mean ["Classic" vs "YAML"](https://learn.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops)?
If so - it doesn't matter. It's just the way how DevOps engineer works with Azure DevOps while defining pipelines.
The code of #adftools is exactly the same. The only nuance is that in YAML you can define precise version of the task, whereas in Classic UI you can define only the major part of version (1.* or 2.*)

**Reply by dvartanian** [reply]
We've just made the switch to federated service connections and so I too had to update my yaml to use the v2 version. Unfortunately, I'm now getting an error:
```
##[error]You cannot call a method on a null-valued expression.
```

This appears to be happen when it attempts to get the federated service connection's token:
```
##[debug]Get-VstsFederatedToken : You cannot call a method on a null-valued expression.
##[debug]At C:\a\_tasks\PublishADFTask_1af843b5-35a0-411f-9a18-9eb7a59fb8b8\2.0.1514\ps_modules\VstsAzureHelpers_\InitializeAzModuleFunctions.ps1:327 char:31
##[debug]+ ... ertionJwt = Get-VstsFederatedToken -serviceConnectionId $connectedSer ...
##[debug]+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##[debug]    + CategoryInfo          : InvalidOperation: (:) [Get-VstsFederatedToken], RuntimeException
##[debug]    + FullyQualifiedErrorId : InvokeMethodOnNull,Get-VstsFederatedToken
```

I have supplied that service connection name in the task inputs as shown below:
![image](https://github.com/user-attachments/assets/9687cd70-e61d-41b5-8604-ab56f1ffb24c)
Is there something else I'm missing that might be causing this?

**Reply by NowinskiK** [reply]
@dvartanian, can you please create a new issue for this? This looks like a problem with your Azure DevOps Service Connection using workload identity federation. Try to run any other PowerShell command checking access to Azure Subscription under this connection. 
For example: `Get-AzContext` or `Get-AzSubscription` and see what you'll get.

**Reply by NowinskiK** [reply]
Check also if you have all PS modules up to date: https://github.com/Azure-Player/azure.datafactory.devops/issues/170
